joining/src/main/scala/akka/cluster/bootstrap/dns/HeadlessServiceDnsBootstrap.scala

```diff
@@ -0,0 +1,278 @@
+
+      context become bootstraping(serviceName, sender())
+  }
+
+  /** In process of searching for seed-nodes */
+  def bootstraping(serviceName: String, replyTo: ActorRef): Receive = {
+    case Internal.AttemptResolve(name, timeout) ⇒
+      resolve(name, timeout).pipeTo(self)
+
+    case DnsResolvedTo(name, contactPoints) ⇒
+      onContactPointsResolved(name, contactPoints)
+
+    case ex: Failure ⇒
+      log.warning("Resolve attempt failed! Cause: {}", ex.cause)
+      scheduleNextResolve(serviceName, settings.dnsResolveTimeout)
+
+    case ObtainedHttpSeedNodesObservation(infoFromAddress, observedSeedNodes) ⇒
+      log.info("Contact point [{}] returned [{}] seed-nodes [{}], initiating cluster joining...", infoFromAddress,
+        observedSeedNodes.size, observedSeedNodes)
+
+      replyTo ! "JOINING SEED NODES!" // FIXME nicer typed message
+
+      val seedNodesList = observedSeedNodes.toList
+      cluster.joinSeedNodes(seedNodesList)
+
+      // once we issued a join bootstraping is completed
+      context.stop(self)
+
+    case NoSeedNodesObtainedWithinDeadline(contactPoint) ⇒
+      log.info(
+          "Contact point [{}] exceeded stable margin with no seed-nodes in sight. " +
+          "Considering weather this node is allowed to JOIN itself to initiate a new cluster.",
+          contactPoint) // TODO debug
+
+      onNoSeedNodesObtainedWithinStableDeadline(contactPoint)
+  }
+
+  private def onContactPointsResolved(serviceName: String, contactPoints: immutable.Seq[String]): Unit = {
+    val newObservation = DnsServiceContactsObservation(timeNow(), contactPoints.toList)
+    lastContactsObservation = lastContactsObservation.sameOrChanged(newObservation)
+
+    if (contactPoints.size < settings.requiredContactPointsNr)
+      onInsufficientContactPointsDiscovered(serviceName, lastContactsObservation)
+    else
+      onSufficientContactPointsDiscovered(serviceName, lastContactsObservation)
+  }
+
+  private def onInsufficientContactPointsDiscovered(serviceName: String,
+                                                    observation: DnsServiceContactsObservation): Unit = {
+    log.info("Discovered ({}) observation, which is less than the required ({}), retrying (interval: {})",
+      observation.observedContactPoints.size, settings.requiredContactPointsNr, settings.dnsResolveTimeout)
+
+    scheduleNextResolve(serviceName, settings.dnsResolveTimeout)
+  }
+
+  private def onSufficientContactPointsDiscovered(serviceName: String,
+                                                  observation: DnsServiceContactsObservation): Unit = {
+    log.info("Initiating contact-point probing, sufficient contact points: {}",
+      observation.observedContactPoints.mkString(", "))
+
+    observation.observedContactPoints.foreach { contactPoint ⇒
+      // TODO would be nice if we can discover this via DNS as well? and leave option to override here
+      val targetPort = settings.httpContactPointPort
```

Comment:
@ktoso - we can discover the target port when we received the `SrvResolved` message from `akka-dns`:

```
object AsyncDnsResolver {
  ...
  case class SrvResolved(name: String, srv: immutable.Seq[SRVRecord])
}

case class SRVRecord(override val name: String, override val ttl: Int,
                     priority: Int, weight: Int, port: Int, target: String) extends ResourceRecord(name, ttl, RecordType.SRV.id.toShort, RecordClass.IN.id.toShort) {
...
}
```

I can see these options in relation to obtaining the value of the port comes from:

1. Only accept SRV record as this would have the actual port number

This means the normal DNS A-record (non-SRV) is ignored/not accepted as it only has address without the port information. This option also means the Akka cluster won't be formed unless there are SRV record present for the seed nodes.

2. Use the port from SRV record if exists, otherwise falling back using A-Record with `settings.httpContactPoint`

This means the DNS A-record (non-SRV) is also accepted with the port value coming from `settings.httpContactPoint`. Maybe have config like `resolve-from-dns-a-record=true`.

As Kubernetes and DCOS both have DNS SRV support, they only need Option 1, and not Option 2.

I was initially thinking about non orchestration deployment, but then AWS Route 53 and GCP Cloud DNS also support SRV records, so perhaps Option 2 is not immediately required.